tools/kagami/src/swarm.rs

```diff
@@ -0,0 +1,1314 @@
+    #[arg(long, short)]
+    peers: NonZeroU16,
+    /// Target directory where to place generated files.
+    ///
+    /// If the directory is not empty, Kagami will prompt it's re-creation. If the TTY is not
+    /// interactive, Kagami will stop execution with non-zero exit code. In order to re-create
+    /// the directory anyway, pass `--outdir-force` flag.
+    #[arg(long)]
+    outdir: PathBuf,
+    /// Re-create the target directory if it already exists.
+    #[arg(long)]
+    outdir_force: bool,
+    /// Do not create default configuration in the `<outdir>/config` directory.
+    ///
+    /// Default `config.json`, `genesis.json` and `validator.wasm` are generated and put into
+    /// the `<outdir>/config` directory. That directory is specified in the Docker Compose
+    /// `volumes` field.
+    ///
+    /// If you don't need the defaults, you could set this flag. The `config` directory will be
+    /// created anyway, but you should put the necessary configuration there by yourself.
+    #[arg(long)]
+    no_default_configuration: bool,
+    /// Might be useful for deterministic key generation.
+    ///
+    /// It could be any string. Its UTF-8 bytes will be used as a seed.
+    #[arg(long, short)]
+    seed: Option<String>,
+}
+
+impl Args {
+    pub fn run(self) -> Outcome {
+        let ui = UserInterface::new();
+
+        let prepare_dir_strategy = if self.outdir_force {
+            PrepareDirectoryStrategy::ForceRecreate
+        } else {
+            PrepareDirectoryStrategy::Prompt
+        };
+        let source = ImageSource::from(self.source);
+        let target_dir = TargetDirectory::new(AbsolutePath::absolutize(self.outdir)?);
+
+        if let EarlyEnding::Halt = target_dir
+            .prepare(&prepare_dir_strategy, &ui)
+            .wrap_err("failed to prepare directory")?
+        {
+            return Ok(());
+        }
+
+        let config_dir = AbsolutePath::absolutize(target_dir.path.join(DIR_CONFIG))?;
+
+        let source = source
+            .resolve(&target_dir, &ui)
+            .wrap_err("failed to resolve the source of image")?;
+
+        let ui = if self.no_default_configuration {
+            PrepareConfigurationStrategy::GenerateOnlyDirectory
+        } else {
+            PrepareConfigurationStrategy::GenerateDefault
+        }
+        .run(&config_dir, ui)
+        .wrap_err("failed to prepare configuration")?;
+
+        DockerComposeBuilder {
+            target_dir: target_dir.path.clone(),
+            config_dir,
+            source,
+            peers: self.peers,
+            seed: self.seed.map(String::into_bytes),
+        }
+        .build()
+        .wrap_err("failed to build docker compose")?
+        .write_file(&target_dir.path.join(FILE_COMPOSE))
+        .wrap_err("failed to write compose file")?;
+
+        ui.log_complete(&target_dir.path);
+
+        Ok(())
+    }
+}
+
+#[derive(ClapArgs, Clone, Debug)]
+#[group(required = true, multiple = false)]
+struct ImageSourceArgs {
```

Comment:
No, Docker Compose works with images. If you pass `image`, it uses it as-is. If you pass `build`, it builds an image with `Dockerfile` at a specified pass, and then works with that image. So, from the perspective of the docker compose, it is a source of the image it will work with.